libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -2123,6 +2023,117 @@ fn update_payment_expirations(
     Ok(payments_res)
 }
 
+impl TryFrom<ListsendpaysPayments> for SendPay {
+    type Error = NodeError;
+
+    fn try_from(value: ListsendpaysPayments) -> std::result::Result<Self, Self::Error> {
+        Ok(SendPay {
+            created_index: value
+                .created_index
+                .ok_or(NodeError::generic("missing created index"))?,
+            updated_index: value.updated_index,
+            groupid: value.groupid,
+            partid: value.partid,
+            payment_hash: value.payment_hash,
+            status: value.status.try_into()?,
+            amount_msat: value.amount_msat.map(|a| a.msat),
+            destination: value.destination,
+            created_at: value.created_at,
+            amount_sent_msat: value.amount_sent_msat.map(|a| a.msat),
+            label: value.label,
+            bolt11: value.bolt11,
+            description: value.description,
+            bolt12: value.bolt12,
+            payment_preimage: value.payment_preimage,
+            erroronion: value.erroronion,
+        })
+    }
+}
+
+impl TryFrom<i32> for SendPayStatus {
+    type Error = NodeError;
+
+    fn try_from(value: i32) -> std::result::Result<Self, Self::Error> {
+        match value {
+            0 => Ok(Self::Pending),
+            1 => Ok(Self::Failed),
+            2 => Ok(Self::Complete),
+            _ => Err(NodeError::generic("invalid send_pay status")),
+        }
+    }
+}
+
+impl TryFrom<SendPayAgg> for Payment {
+    type Error = NodeError;
+
+    fn try_from(value: SendPayAgg) -> std::result::Result<Self, Self::Error> {
+        let ln_invoice = value
+            .bolt11
+            .as_ref()
+            .ok_or(InvoiceError::generic("No bolt11 invoice"))
+            .and_then(|b| parse_invoice(b));
+
+        // For trampoline payments the amount_msat doesn't match the actual
+        // amount. If it's a trampoline payment, take the amount from the label.
+        let (payment_amount, client_label) =
+            serde_json::from_str::<PaymentLabel>(&value.label.clone().unwrap_or_default())
+                .ok()
+                .and_then(|label| {
+                    label
+                        .trampoline
+                        .then_some((label.amount_msat, label.client_label))
+                })
+                .unwrap_or((value.amount.unwrap_or_default(), value.label));
+        let fee_msat = match value.amount {
+            Some(amount) => value.amount_sent.saturating_sub(amount),
+            None => 0,
+        };
+        let status = if value.state & 2 > 0 {
+            PaymentStatus::Complete
+        } else if value.state & 1 > 0 {
+            PaymentStatus::Pending
+        } else {
+            PaymentStatus::Failed
+        };
```

Comment:
I'm not sure this works. From https://github.com/breez/breez-sdk-greenlight/pull/1095/files#diff-7466a53546c6cded460ef5502a9168b13cc930d94fdf7d0ea8166bb10c75fe7eR843-R858, here are the cases:
- `send_pay.status` == `Pending` (0) => `Payment.state` = 0 + 1 = 1
- `send_pay.status` == `Failed` (1) => `Payment.state` = 1 + 4 = 5
- `send_pay.status` == `Complete` (2) => `Payment.state` = 2 + 2 = 4

When we go ahead and decode the state, we get:
- `Payment.state` == 1 => 1 & 2 (=0), 1 & 1 = 1 => `Pending` OK
- `Payment.state` == 5 => 5 & 2 (=0), 5 & 1 = 1 => `Pending` WRONG (should be Failed)
- `Payment.state` == 4 => 4 & 2 (=0), 4 & 1 (=0) => `Failed` WRONG (should be Complete)

By this then follows below that we retrieve the invoice amount rather than the trampoline amount, which may not be present and defaults to 0.